### PR TITLE
Add Related persons to patient registration

### LINKED ISSUE #184: Add related person fields to registration questionnaire
Based on the FHIR spec we would like to have related person's details appended to the patient registration process. FHIR spec for a related person link [here](https://www.hl7.org/fhir/relatedperson.html)

Related person details to be capture 

1. Family name
2. First name
3. Relationship to patient 
4. Contact
a. telecom (phone number)
b. telecom (alternative phone number)

For this 
1. The registration details can be bundled i.e. Patient resource and Related person resource 
2. The resources can be extracted and saved as separate resources upon save/submit 

Subtasks 
- [ ] Edit related person details when editing patient details related to #79 tracked on issue #185

cc. @syedowaisali90 Also liaise with @shoaibmushtaq25 on issue #183 as he also incorporates some of the styling based on  shared designs on #fhirdevs channel

Attached mock
Screen Shot 2021-05-31 at 12 12 31 PM

### PR DESCRIPTION
**Pull request by ekigamba**
Fixes #184

**Type**
Choose one: Feature

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app 

Depends on #356

### CODE REVIEW

File: android/app/src/test/java/org/smartregister/fhircore/StructureMapUtils.kt [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+    val contextR4 = SimpleWorkerContext.fromPackage(pcm.loadPackage("hl7.fhir.r4.core", "4.0.1"))
+    contextR4.isCanRunWithoutTerminology = true
+
+    val scu = org.hl7.fhir.r4.utils.StructureMapUtilities(contextR4)
+    val map = scu.parse(fhirMapToConvert, "PatientRegistration")
+
+    val iParser: IParser = FhirContext.forR4().newJsonParser()
+    val mapString = iParser.encodeResourceToString(map)
+
+    System.out.println(mapString)
```
Comment (pld) [inline android/app/src/test/java/org/smartregister/fhircore/StructureMapUtils.kt:49]: remove?

**Review by pld — APPROVED** [review]

**Comment by pld** [comment]
@ekigamba what's blocking this?

**Comment by ekigamba** [comment]
> @ekigamba what's blocking this?

@pld It's dependent on this PR https://github.com/google/android-fhir/pull/566 being merged and release with the work.

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -30,6 +30,7 @@
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
+        android:usesCleartextTraffic="true"
```
Comment (pld) [inline android/app/src/main/AndroidManifest.xml:33]: do we want this to stay in?

Comment (ekigamba) [reply android/app/src/main/AndroidManifest.xml:33]: Yes, this is a temporary workaround for the server giving http links

**Comment by pld** [comment]
> > @ekigamba what's blocking this?
> 
> @pld It's dependent on this PR [google/android-fhir#566](https://github.com/google/android-fhir/pull/566) being merged and release with the work.

w/566 merged, is this ready for review?

**Comment by ekigamba** [comment]
> > > @ekigamba what's blocking this?
> > 
> > 
> > @pld It's dependent on this PR [google/android-fhir#566](https://github.com/google/android-fhir/pull/566) being merged and release with the work.
> 
> w/566 merged, is this ready for review?

Not yet, we still have another issue blocking further work on this here https://github.com/opensrp/fhircore/issues/398

**Comment by pld** [comment]
> > > > @ekigamba what's blocking this?
> > > 
> > > 
> > > @pld It's dependent on this PR [google/android-fhir#566](https://github.com/google/android-fhir/pull/566) being merged and release with the work.
> > 
> > 
> > w/566 merged, is this ready for review?
> 
> Not yet, we still have another issue blocking further work on this here #398

Looks like 398 is now closed

**Comment by ekigamba** [comment]
> > > > > @ekigamba what's blocking this?
> > > > 
> > > > 
> > > > @pld It's dependent on this PR [google/android-fhir#566](https://github.com/google/android-fhir/pull/566) being merged and release with the work.
> > > 
> > > 
> > > w/566 merged, is this ready for review?
> > 
> > 
> > Not yet, we still have another issue blocking further work on this here #398
> 
> Looks like 398 is now closed

That's correct @pld . I'm currently working on this to get this done

File: android/app/src/test/java/org/smartregister/fhircore/StructureMapPlaygroundUtils.kt [hunk 3]
```diff
@@ -29,8 +29,14 @@ import org.junit.Test
...
 
+/**
+ * Provides a playground for quickly testing and authoring questionnaire and the respective
+ * StructureMap
+ *
+ * This should be removed at a later point once we have a more clear way of doing this
+ */
 @Config(shadows = [FhirApplicationShadow::class])
-class StructureMapUtils : RobolectricTest() {
+class StructureMapPlaygroundUtils : RobolectricTest() {
```
Comment (pld) [inline android/app/src/test/java/org/smartregister/fhircore/StructureMapPlaygroundUtils.kt:39]: what about `StructureMapUtilitiesTest`?

Comment (ekigamba) [reply android/app/src/test/java/org/smartregister/fhircore/StructureMapPlaygroundUtils.kt:39]: It's not a test per se, so I thought the naming and class documentation should help make that clear and prevent any confusion

Comment (pld) [reply android/app/src/test/java/org/smartregister/fhircore/StructureMapPlaygroundUtils.kt:39]: oh I see, these are utilities and fixtures? Then maybe `StructureMapTestUtilities`?

**Review by pld — DISMISSED** [review]

**Review by pld — APPROVED** [review]

**Review by ndegwamartin — APPROVED** [review]

File: android/app/src/test/java/org/smartregister/fhircore/StructureMapPlaygroundUtils.kt (hunk 3 above)
Comment (ekigamba) [reply android/app/src/test/java/org/smartregister/fhircore/StructureMapPlaygroundUtils.kt]: Yea, this change has been made

**Pull request merged**